src/image_ops.rs

```diff
@@ -1582,6 +1584,134 @@ Images are first converted to grayscale. Thresholds should be `>=0` and `< 256`.
         Example::new(1, 1, "thresh 120")
 );
 
+//-----------------------------------------------------------------------------
+// Threshold
+//-----------------------------------------------------------------------------
+
+#[derive(Debug, Copy, Clone, PartialEq, Eq)]
+struct Tile(u32, u32);
+
+impl ImageOp for Tile {
+    fn apply(&self, stack: &mut ImageStack) {
+        let mut image = stack.pop();
+
+        let tiles = tile(&mut image, self.0, self.1);
+
+        for t in tiles.into_iter().rev() {
+            stack.push(t)
+        }
+    }
+
+    // takes 1, returns n
+    fn signature(&self) -> Option<(usize, usize)> {
+        None
+    }
+}
+
+fn tile(mut image: &mut DynamicImage, width: u32, height: u32) -> Vec<DynamicImage> {
+    let max_elem = (image.width() / width + 1) * (image.height() / height + 1);
+    let mut out: Vec<DynamicImage> = Vec::with_capacity(max_elem as usize);
+
+    let background = match image {
```

Comment:
When you've removed the padding (see comment below in reply to your query on the PR) this can go.